### PR TITLE
CORDA-686 - Split Cordapp gradle plugin from cordformation

### PR DESCRIPTION
**Pull request by Clintonio**
Why: Because currently all projects wishing to build nodes or depend on cordapps (tooling for example) must include this plugin and thus become cordapps themselves. It has a slightly invasive effect on the affected gradle setup. 

This also now returns cordformation to it's original purpose and moves us towards gradle plugin stability.

### CODE REVIEW

File: constants.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-gradlePluginsVersion=1.1.0
+gradlePluginsVersion=1.1.1
```
Comment (chrisr3) [inline constants.properties]: Too late, you mean 1.1.2.

File: gradle-plugins/cordapp/build.gradle [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apply plugin: 'java'
```
Comment (chrisr3) [inline gradle-plugins/cordapp/build.gradle]: Isn't this a Groovy plugin?
And where is its `META-INF/gradle-plugins/net.corda.plugins.cordapp.properties` file?

File: gradle-plugins/cordapp/build.gradle [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+description 'Turns a project into a cordapp project that produces cordapp fat JARs'
+
+repositories {
+    mavenCentral()
+    jcenter()
+}
+
+dependencies {
+    gradleApi()
```
Comment (chrisr3) [inline gradle-plugins/cordapp/build.gradle]: You mean this, _surely!?_
```gradle
dependencies {
    compile gradleApi()
    compile localGroovy()
}
```

File: confidential-identities/build.gradle [hunk 4]
```diff
@@ -6,15 +6,14 @@ apply plugin: 'kotlin'
 apply plugin: CanonicalizerPlugin
 apply plugin: 'net.corda.plugins.publish-utils'
 apply plugin: 'net.corda.plugins.quasar-utils'
-apply plugin: 'net.corda.plugins.cordformation'
 apply plugin: 'com.jfrog.artifactory'
```
Comment (chrisr3) [inline confidential-identities/build.gradle]: Isn't `confidential-identities` a CorDapp? So either replace this with `net.corda.plugins.cordapp` or rewrite all of these comments.

File: confidential-identities/build.gradle [hunk 5]
```diff
@@ -6,15 +6,14 @@ apply plugin: 'kotlin'
...
 
 description 'Corda Experimental Confidential Identities'
 
 dependencies {
     // Note the :confidential-identities module is a CorDapp in its own right
     // and CorDapps using :confidential-identities features should use 'cordapp' not 'compile' linkage.
-    cordaCompile project(':core')
+    compile project(':core')
```
Comment (chrisr3) [inline confidential-identities/build.gradle]: I'd have thought that `cordaCompile` was correct for a CorDapp.

File: gradle-plugins/cordapp/src/main/groovy/net/corda/plugins/Utils.groovy [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package net.corda.plugins
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+
+class Utils {
+    public static void createCompileConfiguration(String name, Project project) {
+        if(project.configurations.any { it.name == name }) {
```
Comment (chrisr3) [inline gradle-plugins/cordapp/src/main/groovy/net/corda/plugins/Utils.groovy]: Doesn't this say "Add configuration to project if it's already present?" So maybe instead:
```groovy
if (project.configurations.any { it.name == name }) {
    return
}
Configuration configuration = project.configurations.create(name)
// ...etc
```

File: gradle-plugins/cordapp/src/main/groovy/net/corda/plugins/CordappPlugin.groovy [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                // net.corda or com.r3.corda.enterprise may be a core dependency which shouldn't be included in this cordapp so give a warning
+                if (it.group && (it.group.startsWith('net.corda.') || it.group.startsWith('com.r3.corda.enterprise.'))) {
+                    logger.warn("You appear to have included a Corda platform component ($it) using a 'compile' or 'runtime' dependency." +
+                            "This can cause node stability problems. Please use 'corda' instead." +
+                            "See http://docs.corda.net/cordapp-build-systems.html")
+                } else {
+                    logger.info("Including dependency in CorDapp JAR: $it")
+                }
+            }
+            return filteredDeps.collect { configurations.runtime.files it }.flatten().toSet()
```
[the remark this answers is not in the document]
Comment (Clintonio) [reply gradle-plugins/cordapp/src/main/groovy/net/corda/plugins/CordappPlugin.groovy]: I'm going ot try to statically type this entire file anyway - will make it easier to convert to Java or Kotlin some day.

File: gradle-plugins/cordapp/src/main/groovy/net/corda/plugins/Utils.groovy (hunk 6 above)
Comment (Clintonio) [reply gradle-plugins/cordapp/src/main/groovy/net/corda/plugins/Utils.groovy]: That part was unfinished - I just flipped the conditional to fix it.

File: confidential-identities/build.gradle (hunk 4 above)
Comment (Clintonio) [reply confidential-identities/build.gradle]: Ross has told me it isn't a cordapp and will be merged into core some day, Mike also said the same.

File: confidential-identities/build.gradle (hunk 5 above)
Comment (Clintonio) [reply confidential-identities/build.gradle]: As above; it isn't a cordapp.

File: confidential-identities/build.gradle (hunk 4 above)
Comment (Clintonio) [reply confidential-identities/build.gradle]: Will fix the comments.

File: docs/source/cordapp-build-systems.rst [hunk 8]
```diff
@@ -57,7 +57,7 @@ can find the latest published version of both here: https://bintray.com/r3/corda
 
 ``corda_gradle_plugins_versions`` are given in the form ``major.minor.patch``. You should use the same ``major`` and
 ``minor`` versions as the Corda version you are using, and the latest ``patch`` version. A list of all the available
-versions can be found here: https://bintray.com/r3/corda/cordformation.
+versions can be found here: https://bintray.com/r3/corda/plugins/cordapp.
```
Comment (chrisr3) [inline docs/source/cordapp-build-systems.rst]: Is this URL correct? Why won't it be [https://bintray.com/r3/corda/cordapp](https://bintray.com/r3/corda/cordapp)?

File: gradle-plugins/cordapp/build.gradle [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+description 'Turns a project into a cordapp project that produces cordapp fat JARs'
+
+repositories {
+    mavenCentral()
+    jcenter()
+}
+
+dependencies {
+    compile gradleApi()
+    compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```
Comment (chrisr3) [inline gradle-plugins/cordapp/build.gradle]: Don't you want this instead?:
```gradle
compile "org.jetbrains.kotlin:kotlin-stdlib-jre8:$kotlin_version"
```

**Pull request merged**